jsonargparse_tests/test_typehints.py

```diff
@@ -68,6 +68,20 @@ def test_add_argument_failure_given_type_and_action(parser):
     ctx.match("Providing both type and action not allowed")
 
 
+def test_add_argument_given_type_and_null_action(parser):
+    parser.add_argument("--op1", type=Optional[bool], action=None)
+    assert parser.get_defaults().op1 is None
+
+
+@parser_modes
+def test_add_argument_str_edge_cases_type_and_null_action(parser):
+    parser.add_argument("--val", type=str, action=None)
+    assert parser.parse_args(["--val=e123"]).val == "e123"
+    assert parser.parse_args(["--val=123e"]).val == "123e"
+    val = "1" * 5000
+    assert parser.parse_args([f"--val={val}"]).val == val
+
+
```

Comment:
```suggestion
```

Why this test? Isn't `test_add_argument_given_type_and_null_action` enough to cover the case of action being `None`?

Reply:
Yes, it is enough. I wanted to make sure weird things won't happen but this is not the correct way. It makes a lot of sense to not add it.

Reply:
With `type=None` and `action=None`, there's no problem with the old code. Should we still add the test?